acceptancetests/assess_lxd_space.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+"""Testing spaces and subnets settings for app deployment to lxd container.
+   This is an validation of Bug 1685782:
+   https://bugs.launchpad.net/juju/+bug/1685782
+   The test procedures are based on comment #29.
+"""
+
+from __future__ import print_function
+
+import os
+import sys
+import json
+import logging
+import argparse
+
+from jujucharm import local_charm_path
+from deploy_stack import BootstrapManager
+
+from utility import (
+    JujuAssertionError,
+    add_basic_testing_arguments,
+    configure_logging,
+    )
+
+__metaclass__ = type
+log = logging.getLogger("assess_lxd_space")
+
+
+def assert_initial_spaces(client):
+    """Initial spaces status after bootstrap should show as:
+       no spaces to display"""
+    # Run juju spaces --format json
+    # Bug 1704105: https://bugs.launchpad.net/juju/+bug/1704105
+    # merge_stderr need to be set to True for the initial assert
+    # because spaces list is empty at this point.
+    raw_output = client.get_juju_output(
+        'spaces', '--format', 'json', include_e=False, merge_stderr=True)
+    expected_spaces = 'no spaces to display'
+    if expected_spaces not in raw_output:
+        raise JujuAssertionError(
+            'Incorrect initial spaces status. Found: {}\nExpected: {}'.format(
+            raw_output, expected_spaces))
+    else:
+        log.info('Initial spaces status is: {}.'.format(expected_spaces))
+
+
+def get_subnets(client):
+    # Run juju subnets --format json
```

Comment:
That comment, and similar comments elsewhere, are to be shown to tester and developer to reproduce a bug, because whatever the test does, in bug report, it has to be a reproducible sequence for anyone. From default stdout, even the simplest juju status shows as:
2017-07-27 19:27:37 INFO juju --show-log show-status -m assesslxdspace-20170727191812-temp-env:controller --format yaml
I hardly think it's easy to understand. With --debug setting, the stdout is unreadable, that's the reality. I tried --debug option several times, full of useless info about the controller, model, machines, which I don't really care. It ended up I added print in my code to give special string to debug (Ctrl+F).
Let's say this, when I read our testing code or trying to make a procedure list from a failed test, I really wish I can have such comment to point out what exact commands are.
For log file, I can't remember when the last time I checked it. Do you think anyone else will look at it when he/she trying to make the procedure list, especially when things go wrong? The full focus would be only on console output.
These comments, as I mentioned, are made from user's perspective, as many other places in my code, because I as a user, need it, and is highly like others do too.